src/gui/tray/activitylistmodel.cpp

```diff
@@ -192,31 +193,8 @@ QVariant ActivityListModel::data(const QModelIndex &index, int role) const
         });
     };
 
-    switch (role) {
-    case DisplayPathRole:
-        return getDisplayPath();
-    case PathRole:
-        return QFileInfo(getFilePath()).path();
-    case DisplayLocationRole:
-        return displayLocation();
-    case ActionsLinksRole: {
-        QList<QVariant> customList;
-        foreach (ActivityLink activityLink, a._links) {
-            customList << QVariant::fromValue(activityLink);
-        }
-        return customList;
-    }
-
-    case ActionsLinksContextMenuRole: {
-        return ActivityListModel::convertLinksToMenuEntries(a);
-    }
-    
-    case ActionsLinksForActionButtonsRole: {
-        return ActivityListModel::convertLinksToActionButtons(a);
-    }
-
-    case ActionIconRole: {
-        auto colorIconPath = QStringLiteral("qrc:///client/theme/__COLOR__/"); // We will replace __COLOR__ in QML
+    const auto generateIconPath = [&](bool darkModeIcon) {
```

Comment:
we should really avoid using too much `bool` in such a case
is it possible to quickly replace it with an enum ?

Reply:
Yup, changed it to take into account the current role name in the data method, eliminating the need for an argument in the lambda